helm-charts/basehub/values.yaml

```diff
@@ -1434,21 +1436,6 @@ jupyterhub:
             # casefold group names so we can do case insensitive comparisons.
             groups = {g.name.casefold() for g in spawner.user.groups}
 
-            # If we're using GitHubOAuthenticator, add the user's teams to the groups as well.
-            # Eventually this can be removed, as the user's teams can be set to be groups
-            # once https://github.com/jupyterhub/oauthenticator/pull/735 is merged
-            if isinstance(spawner.authenticator, GitHubOAuthenticator):
-              # Ensure auth_state is populated with teams info
-              auth_state = await spawner.user.get_auth_state()
-              if not auth_state or "teams" not in auth_state:
-                  print(f"User {spawner.user.name} does not have any auth_state set, profile_list filtering not available")
-
-              else:
-                # casefold teams to match what GitHub's API does when doing authorization calls
-                groups |= set([f'{team["organization"]["login"]}:{team["slug"]}'.casefold() for team in auth_state["teams"]])
-
-            print(f"User {spawner.user.name} is part of groups {' '.join(groups)}")
-
             # Filter out profiles with allowed_groups set if the user isn't part of the group
```

Comment:
This, toghether with `GitHubOAuthenticator.auth_state_groups_key` = "teams" and `GitHubOAuthenticator.manage_groups = true` doesn't work unfortunately.

The logs say:
```bash
The auth_state_groups_key teams does not exist in the auth_model. Available keys are: dict_keys(['access_token', 'refresh_token', 'id_token', 'scope', 'token_response', 'github_user', 'teams'])
```
Which is weird.

After printing the auth_state, I also  tried to run https://github.com/jupyterhub/oauthenticator/blob/aadfbf5ff4f25c972bfa03ae65a24057c94f645c/oauthenticator/oauth2.py#L1240-L1248 locally and ran into:

 `TypeError: descriptor 'get' for 'dict' objects doesn't apply to a 'list' object`

Which looks like it's because auth_state[teams] is a list of dicts.

I'm now exploring with the `auth_state_groups_key` as a callable.

cc @yuvipanda

Reply:
I would say for this to pick whatever is the least amount of work that unblocks z24jh 4.0 upgrade!